api_tests/src/test_environment.ts

```diff
@@ -0,0 +1,373 @@
+
+        if (!minerAlreadyRunning) {
+            const tsNode = path.join(
+                this.projectRoot,
+                "api_tests",
+                "node_modules",
+                ".bin",
+                "ts-node"
+            );
+            const minerProgram = path.join(
+                this.projectRoot,
+                "api_tests",
+                "src",
+                "bitcoin_miner.ts"
+            );
+
+            await BitcoinMinerInstance.start(
+                tsNode,
+                minerProgram,
+                path.join(lockDir, "config.json"),
+                minerPidFile,
+                this.logger
+            );
+        }
+
+        this.global.ledgerConfigs.bitcoin = config;
+
+        await release();
+    }
+
+    /**
+     * Start the Ethereum Ledger
+     *
+     * Once this function returns, the necessary configuration values have been set inside the test environment.
+     */
+    private async startEthereum() {
+        const lockDir = await this.getLockDirectory("parity");
+        const release = await ledgerLock(lockDir);
+
+        const parity = await ParityInstance.new(
+            this.projectRoot,
+            path.join(this.logDir, "parity.log"),
+            path.join(lockDir, "parity.pid"),
+            this.logger
+        );
+        const config = await this.startLedger(
+            lockDir,
+            parity,
+            async (parity) => {
+                const rpcUrl = parity.rpcUrl;
+
+                const erc20Wallet = new EthereumWallet(
+                    rpcUrl,
+                    this.logger,
+                    lockDir
+                );
+                const erc20TokenContract = await erc20Wallet.deployErc20TokenContract();
+
+                this.logger.info(
+                    "ERC20 token contract deployed at",
+                    erc20TokenContract
+                );
+
+                return {
+                    rpc_url: rpcUrl,
+                    tokenContract: erc20TokenContract,
+                };
+            }
+        );
+
+        this.global.ledgerConfigs.ethereum = config;
+        this.global.tokenContract = config.tokenContract;
+
+        await release();
+    }
+
+    /**
+     * First starts the Bitcoin and then the Lightning ledgers.
+     *
+     * The Lightning ledgers depend on Bitcoin to be up and running.
+     */
+    private async startBitcoinAndLightning() {
+        await this.startBitcoin();
+
+        // Lightning nodes can be started in parallel
+        await Promise.all([
+            this.startAliceLightning(),
+            this.startBobLightning(),
+        ]);
+
+        await this.setupLightningChannels();
+    }
+
+    private async setupLightningChannels() {
+        const { alice, bob } = this.global.lndWallets;
+
+        await alice.connectPeer(bob);
+
+        await alice.mint({
+            name: AssetKind.Bitcoin,
+            ledger: LedgerKind.Lightning,
+            quantity: "15000000",
```

Comment:
Interesting that opening a channel works without a problem. I assume that the LN library does not add any fees to the transaction. Well, let's leave this as is :)